src/ios/WizPurchase.m

```diff
@@ -67,8 +67,18 @@ - (void)getPendingPurchases:(CDVInvokedUrlCommand *)command {
     [self sendTransactions:[pendingTransactions allValues] toCallback:command.callbackId];
 }
 
+- (void)setApplicationUsername:(CDVInvokedUrlCommand *)command {
+    applicationUsername = [command.arguments objectAtIndex:0];
+
+    CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+}
+
 - (void)purchaseProduct:(SKProduct *)product {
     SKMutablePayment *payment = [SKMutablePayment paymentWithProduct:product];
+    if ([applicationUsername length] != 0) {
```

Comment:
ObjC noob here, forgive me.

Is it not possible for `applicationUsername` to be NULL? Will this not blow up if `setApplicationUsername` has never been called?

Reply:
That's an ObjC quirk yes, `[nil length]` returns 0.
https://stackoverflow.com/questions/899209/how-do-i-test-if-a-string-is-empty-in-objective-c
I'll push a change to ensure `applicationUsername` is always set to `nil` at initialization for non-ARC project.